components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java

```diff
@@ -767,30 +773,60 @@ private RequestPathAuthenticatorConfig getAuthenticatorById(RequestPathAuthentic
         return null;
     }
 
-    private Authenticator buildAuthenticatorResponse(LocalAuthenticatorConfig config) {
+    private Authenticator buildAuthenticatorResponse(LocalAuthenticatorConfig config)
+            throws IdentityApplicationManagementServerException {
 
         Authenticator authenticator = new Authenticator();
         authenticator.setId(base64URLEncode(config.getName()));
         authenticator.setName(config.getName());
         authenticator.setDisplayName(config.getDisplayName());
         authenticator.setIsEnabled(config.isEnabled());
-        authenticator.definedBy(Authenticator.DefinedByEnum.valueOf(config.getDefinedByType().toString()));
         if (config instanceof RequestPathAuthenticatorConfig) {
             authenticator.setType(Authenticator.TypeEnum.REQUEST_PATH);
+            authenticator.setDefinedBy(Authenticator.DefinedByEnum.SYSTEM);
         } else {
             authenticator.setType(Authenticator.TypeEnum.LOCAL);
+            if (AuthenticatorPropertyConstants.DefinedByType.SYSTEM == config.getDefinedByType()) {
+                authenticator.setDefinedBy(Authenticator.DefinedByEnum.SYSTEM);
+                List<AuthenticatorProperty> authenticatorProperties =
+                        Arrays.stream(config.getProperties()).map(propertyToExternal)
+                                .collect(Collectors.toList());
+                authenticator.setProperties(authenticatorProperties);
+            } else {
+                authenticator.setDefinedBy(Authenticator.DefinedByEnum.USER);
+                resolveEndpointConfiguration(authenticator, config);
+            }
```

Comment:
I think it's better if this order is changed.
All authenticators are commonly system. User defined custom authenticators are specifically set. So isn't it better to check for DefinedByEnum.USER in the first condition. else it can be treated as a system authenticator

Reply:
Addressed with commit 4738d610c942a3a655531556cc9dc9283f5caf48